packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -5116,14 +5088,48 @@ abstract class RenderObjectElement extends Element {
     return ancestor as RenderObjectElement;
   }
 
-  ParentDataElement<RenderObjectWidget> _findAncestorParentDataElement() {
+  ParentDataElement<ParentData> _findAncestorParentDataElement() {
     Element ancestor = _parent;
+    ParentDataElement<ParentData> result;
     while (ancestor != null && ancestor is! RenderObjectElement) {
-      if (ancestor is ParentDataElement<RenderObjectWidget>)
-        return ancestor;
+      if (ancestor is ParentDataElement<ParentData>) {
+        result = ancestor as ParentDataElement<ParentData>;
+        break;
+      }
       ancestor = ancestor._parent;
     }
-    return null;
+    assert(() {
+      if (result == null || ancestor == null) {
```

Comment:
Is it ever possible for the result to have been the root?  Or is that condition acceptable?

e.g. if you somehow get to the top most ancestor, and it is the wrong type, we'd have `result != null && ancestor == null` here.

Reply:
I don't fully understand what you mean, if we never find the correct type, result would be null as well?

Also, it is fine to not find any ancestor ParentDataElement.

Reply:
I mean if you find it and it's the root render object

Reply:
Maybe just q comment then about why we are throwing and catching this way would help

Reply:
> I mean if you find it and it's the root render object

You mean, the root of the element tree? That's fine for the purpose of this check. This check only wants to check how many ParentDataElements there are between this RenderObjectElement and the next RenderObjectElement (or the root).

Reply:
Ahh ok